apimachinery: remove pre-apigroups import path logic

sttts opened the pull request.

deads2k commented on pkg/api/meta/install.go:
```diff
@@ -0,0 +1,88 @@
+package meta
```
`pkg/api/legacy`?

sttts replied:
Sure, didn't know where to put it.

deads2k commented on pkg/api/meta/install.go:
```diff
@@ -0,0 +1,88 @@
+	if err := addToCore(scheme); err != nil {
+		glog.Fatal(err)
+	}
+	if err := addToCoreV1(kapi.Scheme); err != nil {
```
Why is this scheme hardcoded?

sttts replied:
it shouldn't

sttts replied:
fixed

deads2k commented on pkg/api/meta/install.go:
```diff
@@ -0,0 +1,88 @@
+func InstallLegacy(group string, addToCore, addToCoreV1 func(*runtime.Scheme) error, groupFactoryRegistry announced.APIGroupFactoryRegistry, registry *registered.APIRegistrationManager, scheme *runtime.Scheme) {
+	groupMetaFactory, found := groupFactoryRegistry[group]
+	if !found {
+		glog.Fatalf("Group %q not found", group)
```
we normally panic, right?

sttts replied:
do we? will change that.

sttts replied:
fixed

deads2k commented:
Minor comments.  After they are fixed up, this lgtm

deads2k commented:
/approve

deads2k commented:
/lgtm

deads2k commented:
/test all

deads2k commented:
@sttts I happened to be near here again today. this is on my wishlist.

enj commented:
The client seems to be defaulting to wrong group:

```
=== BEGIN TEST CASE ===
test/cmd/basicresources.sh:313: executing 'oc run --dry-run foo --image=bar -o "go-template={{.kind}} {{.apiVersion}}"' expecting success and text 'DeploymentConfig v1'
FAILURE after 0.258s: test/cmd/basicresources.sh:313: executing 'oc run --dry-run foo --image=bar -o "go-template={{.kind}} {{.apiVersion}}"' expecting success and text 'DeploymentConfig v1': the output content test failed
Standard output from the command:
DeploymentConfig apps.openshift.io/v1
There was no error output from the command.
=== END TEST CASE ===
```

`DeploymentConfig apps.openshift.io/v1`


```
=== BEGIN TEST CASE ===
test/cmd/newapp.sh:85: executing 'oc new-app nginx -l qwer1345%$$#=self' expecting failure and text 'error: ImageStream "nginx" is invalid'
FAILURE after 0.421s: test/cmd/newapp.sh:85: executing 'oc new-app nginx -l qwer1345%$$#=self' expecting failure and text 'error: ImageStream "nginx" is invalid': the output content test failed
Standard output from the command:
--> Found Docker image b8efb18 (3 weeks old) from Docker Hub for "nginx"

    * An image stream will be created as "nginx:latest" that will track this image
    * This image will be deployed in deployment config "nginx"
    * Port 80/tcp will be load balanced by service "nginx"
      * Other containers can access this service through the hostname "nginx"
    * WARNING: Image "nginx" runs as the 'root' user which may not be permitted by your cluster administrator

--> Creating resources with label  ...
--> Failed

Standard error from the command:
    error: ImageStream.image.openshift.io "nginx" is invalid: metadata.labels: Invalid value: "qwer1345%9070#": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')
    error: DeploymentConfig.apps.openshift.io "nginx" is invalid: [metadata.labels: Invalid value: "qwer1345%9070#": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]'), spec.template.labels: Invalid value: "qwer1345%9070#": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')]
    error: Service "nginx" is invalid: [metadata.labels: Invalid value: "qwer1345%9070#": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]'), spec.selector: Invalid value: "qwer1345%9070#": name part must consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyName',  or 'my.name',  or '123-abc', regex used for validation is '([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9]')]
=== END TEST CASE ===
```

`ImageStream.image.openshift.io "nginx" is invalid`

sttts commented:
@enj thanks for investigating. Have to recheck whether I missed some special case in the legacy code I dropped and replaced by the generic registration.

deads2k commented:
@sttts consistently failing test-cmd

deads2k commented:
I opened  #15955 to fix the compile error and it hit a serialization change in `oc run` and failed test-cmd.
